public/app/core/table_model.ts

```diff
@@ -40,8 +41,19 @@ export default class TableModel {
     this.rows.sort((a, b) => {
       a = a[options.col];
       b = b[options.col];
+
+      if (kbn.isNumeric(a)) {
```

Comment:
I think it would be better to use the column type to decide how to sort values. 
I'm not sure how this would work when strings and numeric values are mixed.